### ISSUE TITLE
Missing autoIncrement property in MySQL

### ISSUE THREAD

**Issue by auwaiyung** [sequelize/sequelize-auto#104]
**"sequelize-auto": "^0.4.17"**

sequelize-auto/lib/dialects/index.js  

Maybe there are something wrong in the function **getForeignKeysQuery**. Missing some condition in Where `C.TABLE_SCHEMA ='" + schemaName + "'`   

```
 exports.mysql = {
  /**
   * Generates an SQL query that returns all foreign keys of a table.
   *
   * @param  {String} tableName  The name of the table.
   * @param  {String} schemaName The name of the schema.
   * @return {String}            The generated sql query.
   */
  getForeignKeysQuery: function(tableName, schemaName) {
    return "SELECT \
        K.CONSTRAINT_NAME as constraint_name \
      , K.CONSTRAINT_SCHEMA as source_schema \
      , K.TABLE_SCHEMA as source_table \
      , K.COLUMN_NAME as source_column \
      , K.REFERENCED_TABLE_SCHEMA AS target_schema \
      , K.REFERENCED_TABLE_NAME AS target_table \
      , K.REFERENCED_COLUMN_NAME AS target_column \
      , C.extra \
      FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE AS K \
      LEFT JOIN INFORMATION_SCHEMA.COLUMNS AS C \
        ON C.TABLE_NAME = K.TABLE_NAME AND C.COLUMN_NAME = K.COLUMN_NAME \
      WHERE \
        K.TABLE_NAME = '" + tableName + "' \
        AND K.CONSTRAINT_SCHEMA = '" + schemaName + "
' AND C.TABLE_SCHEMA ='" + schemaName + "';";
  },
  /**
   * Determines if record entry from the getForeignKeysQuery
   * results is an actual foreign key
   *
   * @param {Object} record The row entry from getForeignKeysQuery
   * @return {Bool}
   */
  isForeignKey: function(record) {
    return _.isObject(record) && _.has(record, 'extra') && record.extra !== "auto_increment";
  },
  /**
   * Determines if record entry from the getForeignKeysQuery
   * results is an actual primary key
   *
   * @param {Object} record The row entry from getForeignKeysQuery
   * @return {Bool}
   */
  isPrimaryKey: function(record) {
    return _.isObject(record) && _.has(record, 'constraint_name') && record.constraint_name === "PRIMARY";
  },
  /**
   * Determines if record entry from the getForeignKeysQuery
   * results is an actual serial/auto increment key
   *
   * @param {Object} record The row entry from getForeignKeysQuery
   * @return {Bool}
   */
  isSerialKey: function(record) {
    return _.isObject(record) && _.has(record, 'extra') && record.extra === "auto_increment";
  }
}
```

**Comment by durango** [comment]
What version of MySQL

**Comment by auwaiyung** [comment]
mysql  Ver 14.14 Distrib 5.7.12, for Win64 (x86_64)

**Comment by sdemurjian** [comment]
missing autoIncrement in MSSQL as well on SQL Server 10.50.6220.0 for fields like

`id[int] NOT NULL IDENTITY(1, 1) PRIMARY KEY,`

**Comment by durango** [comment]
Works OK for me, what does your schema look like?

**Comment by durango** [comment]
Does removing this part `C.TABLE_SCHEMA ='" + schemaName + "'` fix the issue for you?

**Comment by auwaiyung** [comment]
Maybe I have many databases. Some table in different databases, but it have the same table name.

Here is an example:

Database A with table users, 
`CREATE TABLE `users` (
	`id` int(11) UNSIGNED NOT NULL AUTO_INCREMENT, 
	`name` varchar(100) DEFAULT NULL, 
	PRIMARY KEY (`id`)
) `

Database B with table users (same table name)
`CREATE TABLE `users` (
	`id` int(11) UNSIGNED NOT NULL, 
	`name` varchar(100) DEFAULT NULL, 
	PRIMARY KEY (`id`),
) `

`getForeignKeysQuery: function(tableName, schemaName) {
    return "SELECT \
        K.CONSTRAINT_NAME as constraint_name \
      , K.CONSTRAINT_SCHEMA as source_schema \
      , K.TABLE_SCHEMA as source_table \
      , K.COLUMN_NAME as source_column \
      , K.REFERENCED_TABLE_SCHEMA AS target_schema \
      , K.REFERENCED_TABLE_NAME AS target_table \
      , K.REFERENCED_COLUMN_NAME AS target_column \
      , C.extra \
      , C.COLUMN_KEY AS column_key \
      FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE AS K \
      LEFT JOIN INFORMATION_SCHEMA.COLUMNS AS C \
        ON C.TABLE_NAME = K.TABLE_NAME AND C.COLUMN_NAME = K.COLUMN_NAME \
      WHERE \
        K.TABLE_NAME = '" + tableName + "' \
        AND K.CONSTRAINT_SCHEMA = '" + schemaName + "';";
  },`

This sql will become
`SELECT K.CONSTRAINT_NAME AS constraint_name, K.CONSTRAINT_SCHEMA AS sourc
e_schema, K.TABLE_SCHEMA AS source_table, K.COLUMN_NAME AS source_column, K.REFE
RENCED_TABLE_SCHEMA AS target_schema, K.REFERENCED_TABLE_NAME AS target_table, K
.REFERENCED_COLUMN_NAME AS target_column, C.extra, C.COLUMN_KEY AS column_key
    FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE AS K
    LEFT JOIN INFORMATION_SCHEMA.COLUMNS AS C ON C.TABLE_NAME = K.TABLE_NAME
    AND C.COLUMN_NAME = K.COLUMN_NAME
    WHERE K.TABLE_NAME =  "users"
    AND K.CONSTRAINT_SCHEMA =  "b"`

And the result following:

`*************************** 1. row ***************************
constraint_name: PRIMARY
  source_schema: b
   source_table: b
  source_column: id
  target_schema: NULL
   target_table: NULL
  target_column: NULL
          extra: auto_increment
     column_key: PRI
*************************** 2. row ***************************
constraint_name: PRIMARY
  source_schema: b
   source_table: b
  source_column: id
  target_schema: NULL
   target_table: NULL
  target_column: NULL
          extra:
     column_key: PRI
*************************** 3. row ***************************
constraint_name: PRIMARY
  source_schema: b
   source_table: b
  source_column: id
  target_schema: NULL
   target_table: NULL
  target_column: NULL
          extra: auto_increment
     column_key: PRI
3 rows in set (0.09 sec)`

Which I want the table "users" in database B is without the "auto_increment", But this Sql will return extra is "auto_increment".

**Comment by FredLackeyOfficial** [comment]
Take a look at [`file-line-replacer`](https://github.com/FredLackey/file-line-replacer) when you have a moment.  I've had the same problem for years and cannot stand dealing with writing scripts to tweak the final files.  I created [`file-line-replacer`](https://github.com/FredLackey/file-line-replacer) to make fixing these files easier.  This allows sequelize-auto to be used, without issue, and then adjust the files when you're finished.

Here are the steps I use.  First, install it (I prefer installing it globally)...

`npm i -g file-line-replacer`

... and then, from inside of the project using sequelize, I run ... 

```
file-line-replacer \
  --search-dir "/Users/flackey/my-project/src/data/models" \
  --backup-dir "/Users/flackey/my-project/_backup" \
  --old-lines "allowNull: false,|primaryKey: true" \
  --new-lines "autoIncrement: true,|primaryKey: true" \
  --overwrite
``` 
The above command will tweak your models, adding the `autoIncrement` key, and create a backup for each file before it is modified.

**Comment by steveschmitt** [comment]
Fixed in `sequelize-auto@0.5.1`.
